src/test/java/com/mongodb/kafka/connect/source/StartedMongoSourceTaskTest.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.mongodb.kafka.connect.source;
+
+import static java.util.Collections.emptyMap;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.atLeastOnce;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.kafka.connect.source.SourceTaskContext;
+import org.apache.kafka.connect.storage.OffsetStorageReader;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentCaptor;
+
+import com.mongodb.client.ChangeStreamIterable;
+import com.mongodb.client.MongoChangeStreamCursor;
+import com.mongodb.client.MongoClient;
+import com.mongodb.client.model.changestream.FullDocumentBeforeChange;
+
+import com.mongodb.kafka.connect.source.statistics.JmxStatisticsManager;
+
+final class StartedMongoSourceTaskTest {
+  static final class FullDocumentBeforeChangeTest {
```

Comment:
I am proposing the idea of grouping unit tests in different static nested classes by functionality, as this prevents cluttering the test state, set up and tear down methods. If you think this is something that should not be done, or that it may be done, but only when we have unit-tests for different parts of functionality, or something else, please express your thoughts.

Reply:
Anything that validates the connector and makes the code easier to read / come back to / consume sounds great to me. Do you want to add a separate ticket for this?

Reply:
I don't think we need to go and update existing tests, but rather I was curious whether it is fine to start using the approach when writing new tests (and the answer appears to be "yes"), if that seems helpful. As for the existing tests, we may regroup them when we touch them if, again, that seems beneficial in a specific case.